odoo/upgrade_code/18.1-00-odoo-api-superuser-id.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+import re
+import typing
+
+if typing.TYPE_CHECKING:
+    from odoo.cli.upgrade_code import FileManager
+
+
+def upgrade(file_manager: FileManager):
+    import_re = re.compile(r'from odoo import(.*?SUPERUSER_ID.*?)$', re.MULTILINE)
+    usage_re = re.compile(r'odoo.SUPERUSER_ID')
+
+    for file in file_manager:
+        if file.path.suffix != '.py' or file.path.parent.name == 'upgrade_code':
+            continue
+        content = file.content
+        content = usage_re.sub('odoo.api.SUPERUSER_ID', content)
```

Comment:
Yes and no :)

I agree that it may need additional manual adaptation.

This PR handles all occurrences in the framework already. So it will apply to addons, which are executed only once the ORM is initialized, so `odoo.api` will be available. Code upgrades may not handle the migration of 100% of the cases too for simplicity - and given the previous note, this is an edge case.